### DISCUSSION TITLE
How do I use multiple "uses" in a named step

### DISCUSSION THREAD

**Discussion by dapperdandev** [community/community#26573] [Actions]
<p>Try to set the npm version from the the reference that triggers the actions with the following script</p>
<pre><code class="lang-auto">jobs:
  version:
    runs-on: ubuntu-latest
    steps:
      - name: Set Result
        uses: actions/github-script@v3
        id: set-result
        with:
          script: return context.ref.replace('/refs/tags/', '')
          result-encoding: string
      - name: Set Version
         uses: actions/checkout@v2 # error
         uses: actions/setup-node@v2 # error
         with:
           node-version: '14.x'
      - run: npm version --no-git-tag-version "${{steps.set-result.outputs.result}}"
</code></pre>
<p>Github complains about duplicate key “uses”. I’m clearly missing something. I can do this just fine in an unnamed step:</p>
<pre><code class="lang-auto">  build:
    needs: version
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - uses: actions/setup-node@v2
        with:
          node-version: '14.x'
      - run: npm ci
</code></pre>
<p>How do I fix my Set-Version step so that it can use both actions/checkout and actions/setup-node?</p>

**Comment by mickeygousset** [comment, marked as answer]
<p>Each step can only execute a single action. A “uses” statement is executing an action, such as the Checkout action or the setup-node action.   You can’t combine them into one step like you are doing. They have to be two steps:</p>
<pre><code class="lang-auto">- name: Checkout Code
  uses: actions/checkout@v2

- name: Set Version
  uses: actions/setup-node@v2
</code></pre>

**Comment by dapperdandev** [comment]
<p>Thank you! That clears things up and I got my actions working.</p>
